src/config.c

```diff
@@ -472,9 +472,17 @@ void loadServerConfigFromString(char *config) {
              server.client_max_querybuf_len = memtoll(argv[1],NULL);
         } else if ((!strcasecmp(argv[0],"slaveof") ||
                     !strcasecmp(argv[0],"replicaof")) && argc == 3) {
+            if (!strcasecmp(argv[1], "no") && !strcasecmp(argv[2], "one")) {
+                server.masterhost = NULL;
+                continue;
+            }
             slaveof_linenum = linenum;
```

Comment:
we need to either update the slaveof_linenum, or zero it.
i.e. to confirm to the `replicaof directive not allowed in cluster mode` error.
updating it (what i did) won't even allow "replicaof no one".
if we clear it instead, then "replicaof no one will be allowed (and even reset a previous setting.
```suggestion
            slaveof_linenum = linenum;
            if (!strcasecmp(argv[1], "no") && !strcasecmp(argv[2], "one")) {
                server.masterhost = NULL;
                continue;
            }
```